react/features/lobby/middleware.js

```diff
@@ -81,6 +89,55 @@ StateListenerRegistry.register(
                         })
                     );
                     dispatch(playSound(KNOCKING_PARTICIPANT_SOUND_ID));
+
+                    if (navigator.product !== 'ReactNative') {
+                        let notificationTitle;
+                        let customActionNameKey;
+                        let customActionHandler;
+                        let descriptionKey;
+                        let icon;
+
+                        const knockingParticipants = getKnockingParticipants(getState());
+                        const isParticipantsPaneVisible = getParticipantsPaneOpen(getState());
+                        const firstParticipant = knockingParticipants[0];
+
+                        if (knockingParticipants.length > 1) {
+                            descriptionKey = 'notify.participantsWantToJoin';
+                            notificationTitle = i18n.t('notify.participantsJoined', {
+                                joinedParticipants: knockingParticipants.length
+                            });
+                            icon = NOTIFICATION_ICON.PARTICIPANTS;
+                            customActionNameKey = [ 'notify.viewLobby' ];
+                            customActionHandler = [ () => {
+                                dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+                                dispatch(openParticipantsPane());
+                            } ];
+                        } else {
+                            descriptionKey = 'notify.participantWantsToJoin';
+                            notificationTitle = firstParticipant.name;
+                            icon = NOTIFICATION_ICON.PARTICIPANT;
+                            customActionNameKey = [ 'lobby.admit', 'lobby.reject' ];
+                            customActionHandler = [ () => {
+                                dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+                                dispatch(approveKnockingParticipant(firstParticipant.id));
+                            },
+                            () => {
+                                dispatch(hideNotification(LOBBY_NOTIFICATION_ID));
+                                dispatch(rejectKnockingParticipant(firstParticipant.id));
+                            } ];
+                        }
+                        if (!isParticipantsPaneVisible) {
```

Comment:
Check this before building all those notification objects and return early instead.